sources/api/bootstrap-commands/src/main.rs

```diff
@@ -0,0 +1,315 @@
+/*
+# Bootstrap commands
+
+Bootstrap Commands in Bottlerocket is a feature that empowers users to execute predefined commands
+through the Bottlerocket API. It introduces a new systemd service that facilitates the execution of
+these commands, expanding the functionality of the existing Bottlerocket API.
+
+Currently, Bottlerocket provides a set of actions (such as refresh-updates) that can be accessed
+through the apiserver, typically by running the apiclient in the control container. However, this
+new feature aims to extend the accessibility of these actions by allowing Bottlerocket consumers
+to run them prior to the bootstrap container execution.
+
+#Example:
+Given a bootstrap command called `001-test-bootstrap-commands` with the following configuration:
+
+[settings.bootstrap-commands.001-test-bootstrap-commands]
+commands = [[ "apiclient", "set", "motd=helloworld"]]
+essential = true
+mode = "always"
+
+This would update motd from "Welcome to Bottlerocket!" to "helloworld".
```

Comment:
nit: avoid referring to the default motd value in case it changes:
```suggestion
This would set `/etc/motd` to "helloworld".
```